Sources/VLCMedia.m

```diff
@@ -64,6 +64,54 @@
 /* Notification Messages */
 NSString *const VLCMediaMetaChanged              = @"VLCMediaMetaChanged";
 
+/******************************************************************************
+ * VLC callbacks for streaming.
+ */
+int open_cb(void *opaque, void **datap, uint64_t *sizep) {
+    NSInputStream *stream = (__bridge NSInputStream *)(opaque);
+    
+    // Once a stream is closed, it cannot be reopened.
+    if (stream && stream.streamStatus == NSStreamStatusNotOpen) {
+        [stream open];
+        *datap = opaque;
+        *sizep = UINT64_MAX;
+        return 0;
+    }
+    return stream.streamStatus == NSStreamStatusOpen ? 0 : -1;
+}
+
+ssize_t read_cb(void *opaque, unsigned char *buf, size_t len) {
+    NSInputStream *stream = (__bridge NSInputStream *)(opaque);
+    if (!stream) {
+        return -1;
+    }
+    
+    return [stream read:buf maxLength:len];
+}
+
+int seek_cb(void *opaque, uint64_t offset) {
+    NSInputStream *stream = (__bridge NSInputStream *)(opaque);
```

Comment:
If this is not seakable, just pass NULL instead of seek_cb

Reply:
There is no way to check if stream is seekable or not before opening the stream.